corehq/util/couch_doc_processor.py

```diff
@@ -94,7 +94,6 @@ def __iter__(self):
 
         # save iteration state without offset to signal completion
         self.state.pop("offset", None)
-        self.state["retry"] = {}
```

Comment:
@millerdev not sure about this but I had assumed that when I re-run a migration it should just process the retry docs. Though looking again there is no way for them to get removed from the `retry` list if they get successfully processed.

In a situation where some docs are not successfully processed do you expect that the entire migration get's re-run?